nixos/modules/services/backup/mysql-backup.nix

```diff
@@ -9,7 +9,7 @@ let
   cfg = config.services.mysqlBackup ;
   location = cfg.location ;
   mysqlBackupCron = db : ''
-    ${cfg.period} ${cfg.user} ${mysql}/bin/mysqldump ${if cfg.singleTransaction then "--single-transaction" else ""} ${db} | ${gzip}/bin/gzip -c > ${location}/${db}.gz
+    ${cfg.period} ${cfg.user} ${mysql}/bin/mysqldump -u ${cfg.user} ${if cfg.singleTransaction then "--single-transaction" else ""} ${db} | ${gzip}/bin/gzip -c > ${location}/${db}.gz
```

Comment:
How does authentication works in this case?

Reply:
Unix socket authentication, through the new `services.mysql.ensureUsers` option.

Reply:
Did you test whether `-u` is necessary? In theory it should default to the executing user, which should be changed from root to `${cfg.user}` by cron. But I noticed that it seems to be required to explicitly specify when sudoing as another user than root, so this might be the case with cron as well.

Reply:
Did you test whether -u is necessary? In theory it should default to the executing user, which should be changed from root to ${cfg.user} by cron. But I noticed that it seems to be required to explicitly specify when sudoing as another user than root, so this might be the case with cron as well. ❓